src/ViewComposers/LocationSelectComposer.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tipoff\Locations\ViewComposers;
+
+use Tipoff\Locations\Models\Location;
+use Tipoff\Locations\Services\MarketResolver;
+
+class LocationSelectComposer
+{
+    public function compose($view)
+    {
+        $market = MarketResolver::market();
+        $locations = $market ? $market->locations : Location::all();
+
+        $view->with([
+            'market' => $market,
+            'locations' => $locations,
+        ]);
+    }
+}
```

Comment:
Using a view composer makes it easy to render the `locations::location_select` view from anywhere - like from an exception caused by ambiguous location context.

Reply:
I like this!